### ISSUE TITLE
Aprender tecnologias

### ISSUE THREAD

**Issue by byronkamal** [fga-eps-mds/2019.2-Gymnasteg-Wiki#14]
## Descrição
Nessa Sprint as duplas de pareamento irão implementar um Login utilizando as tecnologias React.js (frontend) e NodeJS (backend) para fins de estudo.

## Tasks
- [x] Implementar API em NodeJS
- [x] Implementar frontend em React.js
- [ ] Testes

**Comment by byronkamal** [comment]
**The MDS team has access to Rocketseat's full-stack course (React Native and NodeJS) to study the technologies**

**Comment by Fealps** [comment]
Dando inicio ao pareamento com @fabmont

**Comment by Felizolinha** [comment]
During last week I pair programmed with @LuizPettengill to teach him Node.js and train my React.js skills. We've created [this separate repository](https://github.com/Felizolinha/treinamento-react) to collaborate on the project and also try out ZenHub, GitKraken and Git-flow.

For the back-end of the login system, we made 4 routes. One for logging in, one for signing up, another one for updating the user's password and one for deleting the account. We used MongoDB with the Mongoose library to store the accounts persistently on the server. The database server was located on a remote server on MongoDB Atlas.

For the front-end we used create-react-app to initialize a react project, then cleaned-up some of the generated code. During this process I teached the basics of HTML and CSS to @LuizPettengill, but didn't have time to teach him React.js itself. Later on I created the Login screen and talked to the Node API we've made to try authenticating.

We couldn't learn and apply tests in time.
